visualization/app/codeCharta/util/fileValidator.spec.ts

```diff
@@ -24,143 +24,181 @@ describe("FileValidator", () => {
 		expect(errors.error.length).toBeGreaterThan(0)
 	}
 
+	function expectFileToThrowResult(file) {
+		expect(() => {
+			validate(file)
+		}).toThrow()
+	}
+
+	function expectFileNotToThrowResult(file) {
+		expect(() => {
+			validate(file)
+		}).not.toThrow()
+	}
+
 	it("API version exists in package.json", () => {
 		expect(require("../../../package.json").codecharta.apiVersion).toEqual("1.1")
 	})
 
-	it("should reject null", () => {
-		const errors = FileValidator.validate(null)
-
-		expectFileToBeInvalid(errors)
-		expect(errors.title).toEqual("Error Loading File")
+	it("should throw on null", () => {
+		expectFileToThrowResult(invalidFile)
+		try {
+			validate(null)
+		} catch (e) {
+			expectFileToBeInvalid(e)
+			expect(e.title).toEqual("Error Loading File")
+		}
```

Comment:
Using `try / catch` in tests is almost always not the ideal way to write a test. Instead, rely upon e.g., `assert.throws()` or `expect(foo).toThrow(error)`

The reason is that we never validate that the function did indeed throw. If it would not throw, we'd miss that! We could in theory add another variable to test for that as well but that's not nice to read.

Reply:
We actually check if this test throws an error. I added two functions checking that:

```
function expectFileToThrowResult(file) {
	expect(() => {
		validate(file)
	}).toThrow()
}

function expectFileNotToThrowResult(file) {
	expect(() => {
		validate(file)
	}).not.toThrow()
}
```

They are called before doing something in the test to ensure, that we actually execute the part in the catch-block.

Reply:
After checking the Jest `expect` API, I now see why you implemented it this way.
Seems like their `toThrow` check is not that powerful and is only able to match the class instance or the error message, not error properties.

Are these tests run in a Node.js context? If so, I recommend to use `assert`. But I guess it's run in the browser?

What error message do these errors have by the way? We always just check for the title property?

Reply:
Tests are run in a node.js context.

You're right. We should add tests for the messages as well. Done.